tests/unit/s2n_crl_test.c

```diff
@@ -102,17 +141,404 @@ int main(int argc, char *argv[])
 
     /* CRL issuer hash is retrieved successfully */
     {
-        uint8_t crl_pem[S2N_MAX_TEST_PEM_SIZE] = { 0 };
-        uint32_t crl_pem_len = 0;
-        EXPECT_SUCCESS(s2n_read_test_pem_and_len(S2N_CRL_ROOT_CRL, crl_pem, &crl_pem_len, S2N_MAX_TEST_PEM_SIZE));
-        DEFER_CLEANUP(struct s2n_crl *crl = s2n_crl_new(), s2n_crl_free);
+        DEFER_CLEANUP(struct s2n_crl *crl = load_test_crl(S2N_CRL_ROOT_CRL), s2n_crl_free);
         EXPECT_NOT_NULL(crl);
-        EXPECT_SUCCESS(s2n_crl_load_pem(crl, crl_pem, crl_pem_len));
 
         uint64_t hash = 0;
         EXPECT_SUCCESS(s2n_crl_get_issuer_hash(crl, &hash));
         EXPECT_TRUE(hash != 0);
     }
 
+    DEFER_CLEANUP(struct s2n_crl *root_crl = load_test_crl(S2N_CRL_ROOT_CRL), s2n_crl_free);
+    EXPECT_NOT_NULL(root_crl);
+
+    DEFER_CLEANUP(struct s2n_crl *intermediate_crl = load_test_crl(S2N_CRL_INTERMEDIATE_CRL), s2n_crl_free);
+    EXPECT_NOT_NULL(intermediate_crl);
+
+    DEFER_CLEANUP(struct s2n_crl *intermediate_revoked_crl = load_test_crl(S2N_CRL_INTERMEDIATE_REVOKED_CRL), s2n_crl_free);
+    EXPECT_NOT_NULL(intermediate_revoked_crl);
+
+    /* Save a list of received X509s for s2n_crl_lookup_context tests */
+    struct crl_lookup_data received_lookup_data = { 0 };
+    DEFER_CLEANUP(struct s2n_x509_validator received_lookup_data_validator, s2n_x509_validator_wipe);
+
+    /* CRL validation succeeds for unrevoked certificate chain */
```

Comment:
These tests are pretty thorough, but I'd suggest at least one self-talk functional test. Async callbacks can be tricky, and it's best to double-check that they function as expected when combined with the rest of the s2n-tls logic.

Reply:
I added revoked/unrevoked self talk tests for server authentication, and also revoked/unrevoked self talk tests for client authentication.